### PR TITLE
[ci] Migrate to the 1ES template

### PR DESCRIPTION
**Pull request by pjcollins**
Context: https://aka.ms/1espt

The build pipeline has been updated to extend the 1ES pipeline template,
which will keep the pipeline up to date with the latest compliance and
security requirements. Compliance tasks and scans will run automatically
as part of artifact upload steps.

### CODE REVIEW

**Comment by pjcollins** [comment]
I think I've finally resolved all of the issues here and have fully working runs across all three of the affected pipelines - Xamarin.Android PR, Xamarin.Android (official), and Xamarin.Android Nightly. This should be ready for review (and merge assuming the latest PR build is green), but we can wait to land it until next week in case there is any unintended fallout.

File: build-tools/automation/yaml-templates/run-sliced-nunit-tests.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@ steps:
 
 - ${{ if eq(parameters.retryFailedTests, 'false') }}:
   # If we aren't using auto-retry logic, then this is just a simple template call
-  - template: run-nunit-tests.yaml
+  - template: /build-tools/automation/yaml-templates/run-nunit-tests.yaml@self
```
Comment (jpobst) [inline build-tools/automation/yaml-templates/run-sliced-nunit-tests.yaml:31]: Do you know why this was required? I didn't have to do this in AndroidX to find templates.

**Review by jpobst — APPROVED** [review]

File: build-tools/automation/yaml-templates/run-sliced-nunit-tests.yaml (hunk 1 above)
Comment (pjcollins) [reply build-tools/automation/yaml-templates/run-sliced-nunit-tests.yaml:31]: I'm not sure what caused it, but I hit a dozen or so trigger failures early on in the migration process due to incomplete template paths or that missing `self` declaration.

**Review by dellis1972 — APPROVED** [review]

File: build-tools/automation/azure-pipelines.yaml [hunk 2]
```diff
@@ -52,665 +56,597 @@ variables:
...
 
-    - template: yaml-templates/fail-on-issue.yaml
+        - script: |
+            echo ##vso[task.setvariable variable=JI_JAVA_HOME]%JAVA_HOME_17_X64%
+            echo ##vso[task.setvariable variable=JAVA_HOME]%JAVA_HOME_17_X64%
+          displayName: set JI_JAVA_HOME, JAVA_HOME to $(JAVA_HOME_17_X64)
 
-- template: yaml-templates/stage-msbuild-tests.yaml
+        - script: echo "##vso[task.prependpath]C:\Windows\System32\WindowsPowerShell\v1.0\"
+          displayName: add powershell to path
```
Comment (jonathanpeppers) [inline build-tools/automation/azure-pipelines.yaml:364]: This seems like it is in my `%PATH%` by default already. Does one of the 1ES build agents not have this?

**Review by jonathanpeppers — APPROVED** [review]
So far everything here looks good to me. Do we need to test the "mega pipeline" for this? Or do we have to merge first?

File: build-tools/automation/azure-pipelines.yaml (hunk 2 above)
Comment (pjcollins) [reply build-tools/automation/azure-pipelines.yaml:364]: For some reason the MAUI integration test job was failing on the new pool until I added this:

> Restored D:\a\_work\1\s\maui\src\DotNet\DotNet.csproj (in 7.3 sec).
  Downloading from "[https://dot.net/v1/dotnet-install.ps1"](https://dot.net/v1/dotnet-install.ps1%22) to "D:\a\_work\1\s\maui\bin\dotnet-install.ps1" (70,702 bytes).
  'powershell' is not recognized as an internal or external command, operable program or batch file.

**Comment by pjcollins** [comment]
> So far everything here looks good to me. Do we need to test the "mega pipeline" for this? Or do we have to merge first?

I think it'll be best to merge first and we can sort out any issues with the megapipeline build separately.

**Pull request merged**